C++/include/algorithm/string/heaps_algorithm.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+    Heap's Algorithm
+    ----------------
+
+    This algorithm generates all possible
+permutations of a collection of n elements.    
+The algorithm minimizes movement: it generates each permutation from the previous one by interchanging a single pair of elements, while the other n-2 remain undisturbed.
+    
+    Time Complexity
+    ---------------
+    O(n!), where n is the number of elements in the collection
+
+    Space Complexity
+    ----------------
+    O(n), where n is the number of elements in the collection
+
+*/
+
+
+#ifndef HEAPS_ALGORITHM_HPP
+#define HEAPS_ALGORITHM_HPP
+
+#include <vector>
+
+using namespace std;
```

Comment:
Instead of `using namespace std` (which brings in the entire std namespace), just bring in what you need, i.e.:

```
using std::vector;
using std::string;
```

You can also just prepend `std::` whenever you use those identifiers, as in: `std::swap(collection[0],...`

Reply:
More info here: https://stackoverflow.com/questions/1452721/why-is-using-namespace-std-considered-bad-practice